sdk/messaging/azservicebus/client.go

```diff
@@ -37,6 +38,10 @@ type Client struct {
 type ClientOptions struct {
 	// TLSConfig configures a client with a custom *tls.Config.
 	TLSConfig *tls.Config
+
+	// NewWebSocketConn is a function that can create a net.Conn for use with websockets.
+	// For an example, see ExampleNewClient_usingWebsockets() function in example_client_test.go.
+	NewWebSocketConn func(ctx context.Context, wssHost string) (net.Conn, error)
```

Comment:
I wonder if this can be abstracted away a bit further so that we can avoid having to specify any params here (will all WS clients need just a context and host?).  What if this was just `WebSocketConn net.Conn` so callers have to create this ahead of time?

Reply:
> (will all WS clients need just a context and host?). 

That's a good question. I only have the one websocket library we use today as the example. One thing I can do is put all the parameters into a struct so we can expand it in the future if needed.

>  What if this was just WebSocketConn net.Conn so callers have to create this ahead of time?

The built-in recovery would need to be able to reconnect (depending on what's needed) so we can't rely on a single instance created beforehand.

Also, requiring them to instantiate it up-front would (potentialy) break the ability for them to use this in the same 'lazy initialized' way as the non-websocket path.

Reply:
That's a good point.

I thought about using a struct but it seems a bit clunky, maybe that's not too bad.

Where does the value for `wssHost` come from?

Reply:
It's formed internally by our code. Technically a user could probably figure it out, so maybe we could omit it but we'd probably still want to pass in an parameters-bag anyways. So it seems like a sensible thing to give them.

The example shows how they'd use it and I think it makes life simpler for them.

Reply:
Updated to pass in an 'args' instead.